tests/Benchmarks/System.Memory/IndexOf.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Numerics;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Jobs;
+
+namespace System.Memory.Benchmarks
+{
+    public class SimdConfig : ManualConfig
+    {
+        public SimdConfig()
+        {
+            // https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md
+            Add(Job.ShortRun.With(new EnvironmentVariable[] { new EnvironmentVariable("COMPlus_FeatureSIMD", "1") }).WithId("SIMD Enabled")); 
+            Add(Job.ShortRun.With(new EnvironmentVariable[] { new EnvironmentVariable("COMPlus_FeatureSIMD", "0") }).WithId("SIMD Disabled"));
+        }
+    }
+
+    [Config(typeof(SimdConfig))]
+    public class IndexOf
+    {
+        static byte[] s_buffer = Enumerable.Repeat(byte.MinValue, 2000).ToArray();
+
+        [GlobalSetup]
+        public void PrintInfo() => Console.WriteLine($"// Vector.IsHardwareAccelerated={Vector.IsHardwareAccelerated}"); // just print the info to make sure the config works
+
+        [Benchmark(OperationsPerInvoke = 6 * 4)] // avoid loop overhead, do manuall loop unrolling
```

Comment:
Do we still need to do this and repeat the operation?

Reply:
@ahsonkhan yes, because the cost of creating Span is a big part of the benchmark for the cases where the searched element is at the beginning of memory buffer. 

``` ini

BenchmarkDotNet=v0.10.14.525-nightly, OS=Windows 10.0.16299.309 (1709/FallCreatorsUpdate/Redstone3)
Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
Frequency=3507503 Hz, Resolution=285.1031 ns, Timer=TSC
.NET Core SDK=2.1.300-rc1-008662
  [Host]        : .NET Core 2.1.0-rc1-26424-03 (CoreCLR 4.6.26424.05, CoreFX 4.6.26424.04), 64bit RyuJIT
  SIMD Disabled : .NET Core 2.1.0-rc1-26424-03 (CoreCLR 4.6.26424.05, CoreFX 4.6.26424.04), 64bit RyuJIT
  SIMD Enabled  : .NET Core 2.1.0-rc1-26424-03 (CoreCLR 4.6.26424.05, CoreFX 4.6.26424.04), 64bit RyuJIT

LaunchCount=1  TargetCount=3  WarmupCount=3  

```
|      Method |           Job |  EnvironmentVariables |   at |       Mean |      Error |    StdDev |
|------------ |-------------- |---------------------- |----- |-----------:|-----------:|----------:|
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |    **0** |   **2.763 ns** |  **0.3031 ns** | **0.0171 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |    0 |   5.815 ns |  2.5243 ns | 0.1426 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |    **1** |   **3.086 ns** |  **0.9012 ns** | **0.0509 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |    1 |   5.997 ns |  6.9594 ns | 0.3932 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |    **2** |   **3.354 ns** |  **1.0120 ns** | **0.0572 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |    2 |   5.776 ns |  2.7741 ns | 0.1567 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |    **4** |   **3.280 ns** |  **0.4101 ns** | **0.0232 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |    4 |   5.710 ns |  2.1890 ns | 0.1237 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |    **8** |   **4.459 ns** |  **0.0775 ns** | **0.0044 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |    8 |   5.941 ns |  0.8559 ns | 0.0484 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |   **16** |   **5.976 ns** |  **1.8004 ns** | **0.1017 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |   16 |   6.020 ns |  3.3568 ns | 0.1897 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** |   **30** |   **8.362 ns** |  **0.2031 ns** | **0.0115 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 |   30 |   6.405 ns |  3.3365 ns | 0.1885 ns |
| **SpanIndexOf** | **SIMD Disabled** | **COMPlus_FeatureSIMD=0** | **1000** | **225.725 ns** | **92.0383 ns** | **5.2003 ns** |
| SpanIndexOf |  SIMD Enabled | COMPlus_FeatureSIMD=1 | 1000 |  32.428 ns |  5.5044 ns | 0.3110 ns |